File: znld-web/src/main/java/com/sybd/znld/onenet/dto/CommandParams.java
package com.sybd.znld.onenet.dto;

import com.sybd.onenet.model.OneNetKey;
import lombok.Getter;
import lombok.Setter;

@Getter @Setter
public class CommandParams{
    public Integer deviceId;
    public String imei;
    public OneNetKey oneNetKey;
    public Short timeout;
    public String command;

    public String toUrlString(){
        return "?imei="+this.imei+"&obj_id="+this.oneNetKey.objId+"&res_id="+this.oneNetKey.resId+"&obj_inst_id="+this.oneNetKey.objInstId+"&timeout="+this.timeout;
    }
}

File: znld-web/src/main/java/com/sybd/znld/onenet/dto/CommandParams.kt
package com.sybd.znld.onenet.dto;

class CommandParams(deviceId: Int, imei: String, objId: Int, resId: Int, objInstId: Int, timeout: Int, var command: String)
    : CommonParams(deviceId, imei, objId, resId, objInstId, timeout){
    constructor(deviceId: Int, imei: String, oneNetKey: OneNetKey ,timeout: Int, command: String)
            : this(deviceId, imei, oneNetKey.objId, oneNetKey.resId, oneNetKey.objInstId, timeout, command)
}
